exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -165,41 +169,75 @@ def _translate_to_prometheus(
         metric_name += self._sanitize(metric.name)
 
         description = metric.description or ""
+
+        metric_family_id = "|".join(
```

Comment:
There may be a better way to produce a metric family id.

Reply:
Hmm im wondering if we should implement this grouping in the `Metric` exporter format (a breaking change) before we send it to the exporter.. Originally, I was thinking the flat representation would be easy to work with but it looks like the OTLP and Prom exporters are having to duplicate a lot of grouping logic.

We could make it look more like the OTLP structure

Reply:
Sorry, what do you mean by `Metric` exporter format? And how is that a breaking change?

Reply:
I mean the `Metric` class and the other class it contains https://github.com/open-telemetry/opentelemetry-python/blob/ebdc1011ed9f4e783fcaee12303bdce7611a328c/opentelemetry-sdk/src/opentelemetry/sdk/_metrics/point.py#L78

It's breaking because it would break the prometheus and OTLP exporter code. So if we want to do it, we should do so before declaring stability

Reply:
I mean the `Metric` class and the other class it contains https://github.com/open-telemetry/opentelemetry-python/blob/ebdc1011ed9f4e783fcaee12303bdce7611a328c/opentelemetry-sdk/src/opentelemetry/sdk/_metrics/point.py#L78

It's breaking because it would break the prometheus and OTLP exporter code. So if we want to do it, we should do so before declaring stability

Reply:
@aabmass maybe can you create an issue for this?